src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -804,6 +838,7 @@
                         (.after (to-timestamp producer_timestamp)))
                 (log/warn (trs "Not replacing catalog for certname {0} because local data is newer." certname))
 
+                ;; TODO: adjust this check to allow a catalog change to only catalog inputs to trigger an update for inputs
```

Comment:
To keep in line with how we store package_inventory, we should likely move this up to the same function that calls `replace_catalog!` (from command.clj I believe?) and add something like

```
(when (seq (:inputs catalog))
  (replace-catalog-inputs! (:inputs catalog)))
```